flyway-core/src/main/java/org/flywaydb/core/internal/database/cockroachdb/CockroachDBSchema.java

```diff
@@ -81,19 +90,32 @@ private boolean doEmptyOnce() throws SQLException {
                     "  WHERE table_schema=?" +
                     "  AND table_type='BASE TABLE'" +
                     ")", name);
+        } else if ( ! hasSchemaSupport ) {
+            return !jdbcTemplate.queryForBoolean("SELECT EXISTS (" +
+                    "  SELECT 1" +
+                    "  FROM information_schema.tables " +
+                    "  WHERE table_catalog=?" +
+                    "  AND table_schema='public'" +
```

Comment:
since the two queries are mostly the same, I wonder if it would be better to build up the query with string concatenation. like:

```
String query = "SELECT EXISTS (" +
  "  SELECT 1" +
  "  FROM information_schema.tables " +
  "  WHERE table_catalog=?";
if (hasSchemaSupport) {
  query += " WHERE table_schema=?";
}
query += "  AND table_type='BASE TABLE'" +
// and so on.
```

Reply:
I had the same thoughts, but then I decided to prefer easy readable code with some redundancy over more complex code without redundancy.